### PR TITLE
[DC-1443]: add patch only mode

### PR DESCRIPTION
**Pull request by sbaldwin-rs**
Adds a patch only mode for running updates against a pre-seeded db

### CODE REVIEW

**Comment by sbaldwin-rs** [comment]
Current dependencies on/for this PR:
* master
  * **PR #18** <a href="https://app.graphite.dev/github/pr/rockset/rockbench/18" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/rockset/rockbench/18?utm_source=stack-comment).

File: main.go [hunk 1]
```diff
@@ -152,35 +152,42 @@ func main() {
...
 			}
-			// TODO: this does not guarantee that the writes have finished
 		}
 	}
 
-	if mode == "patch" {
+	if mode == "both" || mode == "patch" {
+		if mode == "patch" {
+			// must explicitly set number of docs so updates are applied evenly across document keys
+			generator.SetMaxDoc(numDocs)
```
Comment (kwadhwa18) [inline main.go:189]: We should add NUM_DOCS as a strict requirement if PATCH mode is specified

File: main.go [hunk 2]
```diff
@@ -29,15 +29,15 @@ func main() {
 	if !(patchMode == "replace" || patchMode == "add") {
 		panic("Invalid patch mode specified, expecting either 'replace' or 'add'")
 	}
-	if !(mode == "add" || mode == "patch") {
-		panic("Invalid mode specified, expecting 'add' or 'patch'")
+	if !(mode == "add" || mode == "patch" || mode == "both") {
```
Comment (kwadhwa18) [inline main.go:32]: we might have delete workload in the future as well, `both` wouldn't help then. Maybe be explicit and say `add_and_patch`?

**Review by kwadhwa18 — APPROVED** [review]
this is great! minor comments

File: main.go (hunk 1 above)
Comment (sbaldwin-rs) [reply main.go:189]: Yes, good call

File: main.go (hunk 2 above)
Comment (sbaldwin-rs) [reply main.go:32]: Renamed it to `add_then_patch`

**Pull request merged**